src/lib/animation-loop.js

```diff
@@ -0,0 +1,33 @@
+// Replacement for the removed 'raf-loop' dependency.
+// Usage: const loop = createAnimationLoop(cb); loop.start();
+// Provides only start() and stop(), matching existing usage pattern.
+export function createAnimationLoop (cb) {
+  let rafId = null
+  let lastTime = null
+
+  function frame (t) {
+    if (rafId == null) return
+    const dt = lastTime == null ? 0 : (t - lastTime)
+    lastTime = t
+    try { cb(dt) } catch (e) { console.warn('animation loop error:', e) }
+    rafId = requestAnimationFrame(frame)
+  }
+
+  return {
+    start () {
+      if (rafId != null) return
+      if (typeof requestAnimationFrame === 'undefined') return
+      lastTime = null
```

Comment:
could we do  `lastTime = performance.now()` here and then avoid some of the null checking?